### DISCUSSION TITLE
High cpu usage when using ya() function wrapper in bash.

### DISCUSSION THREAD

**Discussion by LinArcX** [sxyazi/yazi#602] [1. Q&A]
As mentioned in the [quick-start section](https://yazi-rs.github.io/docs/quick-start/), for bash, we should use a function wrapper like this:
```
function ya() {
	local tmp="$(mktemp -t "yazi-cwd.XXXXX")"
	yazi "$@" --cwd-file="$tmp"
	if cwd="$(cat -- "$tmp")" && [ -n "$cwd" ] && [ "$cwd" != "$PWD" ]; then
		cd -- "$cwd"
	fi
	rm -f -- "$tmp"
}
```
It causes high CPU usage on my machine.

Also, I don't understand why it creates lots of processes.

![screenshot_2024_01_29_20_43_17](https://github.com/sxyazi/yazi/assets/10884422/a0b57ac2-ec0d-42cd-ae5e-6be42130097e)

**Comment by sxyazi** [comment]
Please do a `which ya` and `which yazi`, then paste the output here
